src/cpp/src/marius.cpp

```diff
@@ -96,6 +96,99 @@ void marius(int argc, char *argv[]) {
         embeddings->unload(true);
         src_rel->unload(true);
         dst_rel->unload(true);
+
+
+        // garbage collect
+        delete trainer;
+        delete train_set;
+        if (will_evaluate) {
+            delete evaluator;
+            delete eval_set;
+        }
+
+        switch (marius_options.storage.edges) {
```

Comment:
this code block seems to be repeated three times -- maybe a good candidate to put inside a util function or something like that?

Reply:
The code blocks are slightly different, gotta handle the storage for the edges, node embedding parameters, and relation embedding parameters a little differently. I tossed these into a util function to make it cleaner.